src/openforms/forms/models/form_step.py

```diff
@@ -70,7 +70,7 @@ class Meta:
         verbose_name_plural = _("form steps")
 
     def __str__(self):
-        return _("Form step {order}").format(order=self.order)
+        return f"{self.form.admin_name} {_('step')} {self.order}: {self.form_definition.admin_name}"
```

Comment:
- Do not use f-strings for `__str__` implementations, the entire template should be marked translatable
- When interpolating related objects, always consider the case that the relation is not set/saved yet. This crashes now for `str(FormStep())`